src/algebra/monoid_algebra/degree.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebra.monoid_algebra.basic
+
+/-!  #  Degree of an `add_monoid_algebra`
+
+Let `R` be a semiring, let `A` be a Type with a linear order.
+
+This file defines the degree of an element of `add_monoid_algebra R A`.  It takes values in
+the grading Type `A` to which we add an element `⊥`, smaller than all the elements of `A`.
+
+Currently, the only result is `degree_mul_le`: the degree of a product is at most the sum of the
+degrees. -/
+
+namespace add_monoid_algebra
+
+variables {R A : Type*} [semiring R] [linear_order A]
+
+/--  Let `R` be a semiring, let `A` be a Type with a linear order, and let `f : R[A]` be an
+an element of `add_monoid_algebra R A`.  The degree of `f` takes values in `with_bot A` and
+it is the maximum of the support of `f` or `⊥`, depending on whether `f` is non-zero or not. -/
+def degree (f : add_monoid_algebra R A) : with_bot A :=
+f.support.max
```

Comment:
If you replace this with `.sup id`, is the lemma below still true in partial orders?

Reply:
Good suggestions!  Except, Lean complains that it cannot find a `semilattice_sup` instance neither on `A`, nor on `with_bot A`.

I used `.sup coe/id`, depending on the target and weakened `linear_order` to `partial_order`, if this is relevant.

Reply:
There is [docs#multiset.sup_add](https://leanprover-community.github.io/mathlib_docs/find/multiset.sup_add) which looks promising!  Except that it would have to be ported to finsets.

Reply:
Sorry, I of course mean "in `semilattice_sup`"

Reply:
Yes, I understood that this is what you meant just after sending my initial reaction!

Reply:
Turns out that the answer is yes, and, unless I messed up with Git, I should have pushed a version that reverted to the original and replaces `max`s with `sup`s.